### PR TITLE
Setting up `23.5.0` branch

### PR DESCRIPTION
**Pull request by zoxpx**
Sets up `23.5.0` branch.

Signed-off-by: Zoran Rajic <zrajic@purestorage.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

note: new PR recreated from https://github.com/libopenstorage/operator/pull/1022

### CODE REVIEW

File: deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml:1]: The CRD files need to be copied from `deploy/crds/`. This CRD schema seems to be old and not up to date.

Use this file - https://github.com/libopenstorage/operator/blob/px-rel-23.5.0/deploy/crds/core_v1_storagecluster_crd.yaml

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM
Just one comment about the CRD schema.

File: deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml
Comment (zoxpx) [reply deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml:1]: BTW, should we also fix the https://github.com/libopenstorage/operator/blob/px-rel-23.4.0/deploy/olm-catalog/portworx/23.4.0/core_v1_storagecluster_crd.yaml  (from `px-rel-23.4.0` branch?)

Comment (piyush-nimbalkar) [reply deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml:1]: @zoxpx this is still not the same as https://github.com/libopenstorage/operator/blob/px-rel-23.5.0/deploy/crds/core_v1_storagecluster_crd.yaml.

Comment (piyush-nimbalkar) [reply deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml:1]: Yes, we should fix that in the `deploy/olm-catalog/portworx/23.4.0` folder as well. Thanks for catching it.

Comment (piyush-nimbalkar) [reply deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml:1]: Once we fix it, we should update the `px-rel-23.4.0` branch as well as master.

Comment (zoxpx) [reply deploy/olm-catalog/portworx/23.5.0/core_v1_storagecluster_crd.yaml:1]: > @zoxpx this is still not the same as https://github.com/libopenstorage/operator/blob/px-rel-23.5.0/deploy/crds/core_v1_storagecluster_crd.yaml.

I should have explained better --
* with [6ae2f0f](https://github.com/libopenstorage/operator/pull/1023/commits/6ae2f0f1e96ae19757af0b1b300c162d4410673f) I've uploaded the new `core_v1_storagecluster_crd.yaml` for `px-rel-23.5.0` branch from the location you pointed out
* I was asking if we should also update `core_v1_storagecluster_crd.yaml` for `px-rel-23.4.0` branch  (that one will probably also be old/obsolete)

**Pull request merged**